scripts/trim-low-abund.py

```diff
@@ -13,71 +13,71 @@
+from khmer.utils import (write_record, write_record_pair, broken_paired_reader)
+
 
 DEFAULT_NORMALIZE_LIMIT = 20
 DEFAULT_CUTOFF = 2
 
-DEFAULT_K = 32
-DEFAULT_N_HT = 4
-DEFAULT_MIN_HASHSIZE = 1e6
-
 # see Zhang et al., http://arxiv.org/abs/1309.2975
 MAX_FALSE_POSITIVE_RATE = 0.8
 
 
-def output_single(read, trim_at=None):
-    name = read.name
-    sequence = read.sequence
-
-    accuracy = None
+def trim_record(read, trim_at):
+    new_read = _screed_record_dict()
+    new_read.name = read.name
+    new_read.sequence = read.sequence[:trim_at]
     if hasattr(read, 'accuracy'):
-        accuracy = read.accuracy
+        new_read.accuracy = read.accuracy[:trim_at]
 
-    if trim_at is not None:
-        sequence = sequence[:trim_at]
-        if accuracy:
-            accuracy = accuracy[:trim_at]
+    return new_read
 
-    if accuracy:
-        return "@%s\n%s\n+\n%s\n" % (name, sequence, accuracy)
-    else:
-        return ">%s\n%s\n" % (name, sequence)
 
+def get_parser():
+    epilog = """
+    The output is one file for each input file, <input file>.abundtrim, placed
+    in the current directory.  This output contains the input sequences
+    trimmed at low-abundance k-mers.
 
-def main():
-    parser = argparse.ArgumentParser(description='XXX')
-
-    env_ksize = os.environ.get('KHMER_KSIZE', DEFAULT_K)
-    env_n_hashes = os.environ.get('KHMER_N_HASHES', DEFAULT_N_HT)
-    env_hashsize = os.environ.get('KHMER_MIN_HASHSIZE', DEFAULT_MIN_HASHSIZE)
-
-    parser.add_argument('--ksize', '-k', type=int, dest='ksize',
-                        default=env_ksize,
-                        help='k-mer size to use')
-    parser.add_argument('--n_hashes', '-N', type=int, dest='n_hashes',
-                        default=env_n_hashes,
-                        help='number of hash tables to use')
-    parser.add_argument('--hashsize', '-x', type=float, dest='min_hashsize',
-                        default=env_hashsize,
-                        help='lower bound on hashsize to use')
+    The ``-V/--variable-coverage`` parameter will, if specified,
+    prevent elimination of low-abundance reads by only trimming
+    low-abundance k-mers from high-abundance reads; use this for
+    non-genomic data sets that may have variable coverage.
+
+    Note that the output reads will not necessarily be in the same order
+    as the reads in the input files; if this is an important consideration,
+    use ``load-into-counting.py`` and ``filter-abund.py``.  However, read
+    pairs will be kept together, in "broken-paired" format; you can use
+    ``extract-paired-reads.py`` to extract read pairs and orphans.
+
+    Example::
+
+        trim-low-abund.py -x 5e7 -k 20 -C 2 data/100k-filtered.fa
+    """
+
+    parser = build_counting_args(
+        descr='Trim low-abundance k-mers using a streaming algorithm.',
+        epilog=textwrap.dedent(epilog))
 
     parser.add_argument('--cutoff', '-C', type=int, dest='abund_cutoff',
```

Comment:
I rather like it when the command line option and the argument destination are the same (no `dest='...'`). Makes the code a lot easier to read and think about.